plugin/src/main/java/org/opensearch/ml/settings/MLCommonsSettings.java

```diff
@@ -108,6 +108,9 @@ private MLCommonsSettings() {}
             Setting.Property.Dynamic
         );
 
+    public static final Setting<Boolean> ML_COMMONS_REMOTE_INFERENCE_ENABLED = Setting
```

Comment:
I see AD plugin also has setting to disable the whole plugin https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/settings/ADEnabledSetting.java#L35C5-L35C81

Can we also add that for disabling whole plugin?

Reply:
Let's get in this feature flag first given the tight timeline. The plugin level disabling can be added in a different PR later.

Reply:
Let's add details comments about the settings we are adding and the coverage like (to block APIs for connector creation, model registration, model deploy and model predictions)

Reply:
Sure, can you create a github issue to track this?

How about we follow the same naming pattern like AD, for example `"plugins.anomaly_detection.hcad_cold_start_interpolation.enabled"` , the pattern is `plugins.<plugin_name>.<feature_name>.enabled`